activitysim/core/cleaning.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import pandas as pd
+
+from . import inject
+
+
+def recode_to_zero_based(values, mapping):
+    values = np.asarray(values)
+    zone_ids = pd.Index(mapping, dtype=np.int32)
+    if (
+        zone_ids.is_monotonic_increasing
+        and zone_ids[-1] == len(zone_ids) + zone_ids[0] - 1
+    ):
+        offset = zone_ids[0]
+        result = values - offset
+    else:
+        n = len(zone_ids)
+        remapper = dict(zip(zone_ids, pd.RangeIndex(n)))
+        if n < 128:
+            out_dtype = np.int8
+        elif n < (1 << 15):
+            out_dtype = np.int16
+        elif n < (1 << 31):
+            out_dtype = np.int32
+        else:
+            out_dtype = np.int64
+        result = np.fromiter((remapper.get(xi) for xi in values), out_dtype)
+    return result
+
+
+def should_recode_based_on_table(tablename):
+    try:
+        base_df = inject.get_table(tablename).to_frame()
+    except (KeyError, RuntimeError):
+        # the basis table is missing, do not
+        return False
+    if base_df.index.name and f"_original_{base_df.index.name}" in base_df:
+        return True
+    return False
+
+
+def recode_based_on_table(values, tablename):
```

Comment:
Is it valuable for the user to know that recoding happens?  Looks like this function and the one above get called in los.py when loading data and in the init of the MazSkimDict in skim_dictionary.py.  Neither of these places inform the user that any recoding is happening under the hood.  Suggest putting a logging statement in here or where these are called.

Reply:
ok